samples/wifi/sta/src/main.c

```diff
@@ -229,6 +244,16 @@ static int wifi_disconnect(void)
 	return 0;
 }
 
+K_THREAD_DEFINE(client_status,
+		CONFIG_CLIENT_STATUS_THREAD_STACK_SIZE,
+		status_thread,
+		NULL,
+		NULL,
+		NULL,
+		0,
+		0,
+		0);
```

Comment:
0 is no delay, so, thread starts immediately, you should use K_FOREVER as you are starting the thread explicityl